### DISCUSSION TITLE
does stream support the feature like consumer group

### DISCUSSION THREAD

**Discussion by rxjhfmf** [rabbitmq/rabbitmq-server#9434] [Other]
### Is your feature request related to a problem? Please describe.

i use one topic exchange and 5+ classic queues that the queues binding the same routekey used by diffrent service. and  every queue has several thousands published/s, also erver serivce has 5 consumer to consume the message. 
recently , i saw the stream ,after some try,i got 2 questions:
1: I didn't known how to set 5 consumers  in a group ,so that they can work like 5 consumers with a classic queue;
2: when the consumer restarted ,how can i set the offset property to make sure it continues the message after the last confirmed messageid . 

### Describe the solution you'd like

for question one : does stream support the feature like consumer group ;
for question two :  maybe i use the wrong way to declare the consume , so , how to tell the broker , a restarted consumer is the 
 same one

**Comment by michaelklishin** [comment]
Please use Discussions for questions in the future.

@rxjhfmf you can register N consumers on a stream or superstream. Streams allow for repeated consumption, consumed messages are not removed, so the analogy won't hold. If you want the behavior of queues, you can use quorum queues. There is a [feature comparison table](https://rabbitmq.com/streams.html#feature-comparison) in the Streams guide.

How would "the same" (a returning) consumer be different from a newly added one? You can keep track of the stream offset used and re-register at the original offset, otherwise all stream consumers are identical (at most they differ in their activity state if you use [SAC](https://rabbitmq.com/streams.html#single-active-consumer)).

**Reply by rxjhfmf** [reply]
Yes, SAC for Streams can solve it when my service instance restarted , but how can i make the consumers on stream just like on work queues

**Reply by mkuratczyk** [reply]
Do you mean competing consumers? That's not supported for streams right now. https://github.com/rabbitmq/osiris/issues/2

**Reply by rxjhfmf** [reply]
Yes, competing consumers in one group with the same consumer reference , is there any plan

**Reply by mkuratczyk** [reply]
The issue I linked to is "the plan". Nothing more specific than that - the idea is there, but no work at this point.

**Comment by acogoluegnes** [comment]
> competing consumers in one group with the same consumer reference , is there any plan

You can use [super streams](https://blog.rabbitmq.com/posts/2022/07/rabbitmq-3-11-feature-preview-super-streams/) for this, your data is partitioned across several individual streams and a consumer instance is allocated one partition. This is one way to process messages in parallel.
